crates/core/src/subscription/module_subscription_actor.rs

```diff
@@ -85,11 +85,9 @@ impl ModuleSubscriptions {
         // This also makes it possible for `broadcast_event` to get scheduled before the subsequent part here
         // but that should not pose an issue.
         let mut subscriptions = self.subscriptions.write();
-        drop(tx);
```

Comment:
I feel  we do need to `drop(tx)` early for any correction issue, we can do it being more conserative.

dropping subscription lock should be enough as that will also not let the reducers to commit (reducer need subscription read lock to commit), so their `eval_incr` will also not start.

if we not `drop(tx)` early, reducer execution will not happen parallel to `eval`, though right now we are letting execution to happen but not commit (which is also fine).

At the end, it depends whether we want to be more conservative or look for bit more perfomance.

Reply:
We discussed async: Holding the tx for the lifetime of the call does keep a reducer from executing until we push onto the send queue, but that shouldn't have a noticeable impact on throughput, plus it makes testing easier. So I think we're both in favor of this from a correctness standpoint.